pipeline-sync.yml

```diff
@@ -1,60 +1,77 @@
 ---
+docker-credentials: &docker-credentials
+  username: ((ce-dockerhub-id))
+  password: ((ce-dockerhub-token))
+
+repo-authorization: &repo-authorization
+  uri: git@github.com:openstax/((book-repo)).git
+  private_key: ((github-private-key))
+
+resource_types:
+  - name: archive-api
+    type: registry-image
+    source:
+      <<: *docker-credentials
+      repository: bn16/archive-version-checker
+
 resources:
-  - name: every-3-hours
-    type: time
+  - name: version-checker
+    type: archive-api
     source:
-      interval: 3h
+      archive_server: ((archive-server))
+      book_repo: ((book-repo))
+      github_token: ((github-api-token))
 
-  - name: osbook
+  - name: ((book-repo))
     type: git
     source:
       branch: main
-      uri: ((osbook-git-uri))
-      private_key: ((ce-github-private-key))
+      <<: *repo-authorization
 
-  - name: synced-osbook
+  - name: synced-((book-repo))
```

Comment:
Did it turn out that this resource is necessary? Is `put: ((book-repo))` possible?

Reply:
yes, when trying to read and write to the same git resource - i found that the best way was to use two resources. 
if we only use `osbook` and error will occur (I think it could have been handled by running git commands, but this was much cleaner)

Reply:
(the naming of the `put` resource doesn't seem right but..) 
I believe this was the error:
<img width="1273" alt="Screen Shot 2021-03-09 at 1 48 11 PM" src="https://user-images.githubusercontent.com/11208820/111542756-226a9880-8740-11eb-8db0-b155d92ef28f.png">

Reply:
isn't this solved by using `force`, regardless of the resource used to `put`?

Reply:
I think I mis-answered your first question - 
"Did it turn out that this resource is necessary? Is put: ((book-repo)) possible?"
This was necessary with the detail that we were working with more than one branch, and that we would always pull from main branch but not `put` (push) back to the main branch. 

Initially is was unclear having thrown a 'staging/qa' branch in earlier discussions.  I made more assumptions dealing with a `put` back to the a branch that was not `main`. 
 I believe that error did because I wasn't using the `force` param, so that definitely makes sense. 

but (with confirmation/ clarification from check in) in the end, that resource isn't necessary. (I'm removing it!) 
Since we will only be pull and pushing to the main branch. - 

tldr; will be updated to use one git (book) resource